test/integration/user_login_test.rb

```diff
@@ -0,0 +1,25 @@
+require 'test_helper'
+
+class UserLoginTest < ActionDispatch::IntegrationTest
+  include Capybara::DSL
+
+  test "user can login" do
+    user = User.create(username: 'user',
+                       password: 'password',
+                       first_name: 'John',
+                       last_name: 'Doe',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.